tests/gui/test_widgets.py

```diff
@@ -722,15 +723,74 @@ def test_FileWidget_mousePressEvent_open(mocker):
     Should fire the expected open event handler in the logic layer.
     """
     mock_controller = mocker.MagicMock()
+    mock_signal = mocker.MagicMock()  # not important for this test
     source = factory.Source()
     file_ = db.File(source=source, uuid='uuid', size=123, filename='1-mah-reply.gpg',
                     download_url='http://mah-server/mah-reply-url', is_downloaded=True)
 
-    fw = FileWidget(source, file_, mock_controller)
+    fw = FileWidget(source, file_, mock_controller, mock_signal)
     fw.mouseReleaseEvent(None)
     fw.controller.on_file_open.assert_called_once_with(file_)
 
 
+def test_FileWidget_clear_deletes_items(mocker, homedir):
+    """
+    Calling the clear() method on FileWidget should delete the existing items in the layout.
+    """
+    mock_controller = mocker.MagicMock()
+    mock_signal = mocker.MagicMock()  # not important for this test
+    source = factory.Source()
+    file_ = db.File(source=source, uuid='uuid', size=123, filename='1-mah-reply.gpg',
+                    download_url='http://mah-server/mah-reply-url', is_downloaded=True)
+
+    fw = FileWidget(source, file_, mock_controller, mock_signal)
+    assert fw.layout.count() != 0
+
+    fw.clear()
+
+    assert fw.layout.count() == 0
+
+
+def test_FileWidget_on_file_download_updates_items_when_uuid_matches(mocker, homedir):
+    """
+    The _on_file_download method should clear and update the FileWidget
+    """
+    mock_controller = mocker.MagicMock()
+    mock_signal = mocker.MagicMock()  # not important for this test
+    source = factory.Source()
+    file_ = db.File(source=source, uuid='uuid', size=123, filename='1-mah-reply.gpg',
```

Comment:
Unless there's a reason for these specific values, we may want to use `factory.File` to reduce the Checkov's gun effect of naming all these values.